src/colony/modules/_net_proxied.js

```diff
@@ -0,0 +1,200 @@
+    PROXY_HOST = process.env.PROXY_HOST || "proxy.tessel.io",
+    PROXY_PORT = +process.env.PROXY_PORT || 443,
+    PROXY_TRUSTED = +process.env.PROXY_TRUSTED || 0,
+    PROXY_TOKEN = process.env.PROXY_TOKEN || process.env.TM_API_KEY,
+    PROXY_LOCAL = process.env.PROXY_LOCAL || _PROXY_LOCAL,
+    PROXY_IDLE = +process.env.PROXY_IDLE || 90e3,
+    PROXY_CERT = process.env.PROXY_CERT || [
+      "-----BEGIN CERTIFICATE-----",
+      "MIICazCCAdQCCQDr2mJoysZo9DANBgkqhkiG9w0BAQUFADB6MQswCQYDVQQGEwJV",
+      "UzELMAkGA1UECBMCQ0ExETAPBgNVBAcTCEJlcmtlbGV5MQ8wDQYDVQQKEwZUZXNz",
+      "ZWwxGDAWBgNVBAMTD3Byb3h5LnRlc3NlbC5pbzEgMB4GCSqGSIb3DQEJARYRdGVh",
+      "bUB0ZWNobmljYWwuaW8wHhcNMTUwMTIzMDExOTQ5WhcNMTUwMjIyMDExOTQ5WjB6",
+      "MQswCQYDVQQGEwJVUzELMAkGA1UECBMCQ0ExETAPBgNVBAcTCEJlcmtlbGV5MQ8w",
+      "DQYDVQQKEwZUZXNzZWwxGDAWBgNVBAMTD3Byb3h5LnRlc3NlbC5pbzEgMB4GCSqG",
+      "SIb3DQEJARYRdGVhbUB0ZWNobmljYWwuaW8wgZ8wDQYJKoZIhvcNAQEBBQADgY0A",
+      "MIGJAoGBAMBZDd+DvwOxtCHl0tVYjctfoim7GvVrE257vofr7oi1SXGXf1ZPcPgb",
+      "DDiq5zq5G38JaHNHpkUq5+J8XUNXgdITAm0Pj4RHwwIHejmWh7ZWRZqsxrz+E6T0",
+      "aj+RIqe3wAfNmr4N7pZ8un9XL1abcTREOMdaLvsNBLhFSJOf3B2tAgMBAAEwDQYJ",
+      "KoZIhvcNAQEFBQADgYEAbNbBFOe5P33FWUup35d/zUdU/q2j5xDGhMbyCVkdONh9",
+      "QrjD9jStLIhaLOjXleI5S+TT9Vgqknn7D9Q977+/wTAg6aXwSBe1LK/V4du92eSx",
+      "/stxtxpFH7BQ1YPuuzmIcYd8rGPywjV91fsgqmWIZFe7FHGbRqKNKPCeR3Nv4tg=",
+      "-----END CERTIFICATE-----",
+    ].join('\n');
+
+/**
+ * Tunnel helpers
+ */
+
+function createTunnel(cb) {
+  if (_PROXY_DBG) console.log("TUNNEL -> START", new Date());
+  tls.connect({host:PROXY_HOST, port:PROXY_PORT, proxy:false, ca:[PROXY_CERT]}, function () {
+    var proxySocket = this,
+        tunnel = streamplex(streamplex.B_SIDE);
+    tunnel.pipe(proxySocket).pipe(tunnel);
+    proxySocket.on('error', shutdownTunnel);
+    proxySocket.on('close', shutdownTunnel);
+    proxySocket.on('error', cb);
+    
+    var idleTimeout;
+    tunnel.on('inactive', function () {
+      if (_PROXY_DBG) console.log("TUNNEL -> inactive", new Date());
+      idleTimeout = setTimeout(shutdownTunnel, PROXY_IDLE);
+    });
+    tunnel.on('active', function () {
+      if (_PROXY_DBG) console.log("TUNNEL -> active", new Date());
+      clearTimeout(idleTimeout);
+    });
+    
+    tunnel.sendMessage({token:PROXY_TOKEN});
+    tunnel.once('message', function (d) {
+      if (_PROXY_DBG) console.log("TUNNEL: auth response?", d);
+      proxySocket.removeListener('error', cb);
+      if (!d.authed) cb(new Error("Authorization failed."));
+      else cb(null, tunnel);
+    });
+    function shutdownTunnel(e) {
+      if (_PROXY_DBG) console.log("TUNNEL -> STOP", new Date());
+      tunnel.destroy(e);
+      if (this !== proxySocket) proxySocket.end();
+      proxySocket.removeListener('close', shutdownTunnel);
+    }
+  }).on('error', cb);
+}
+
+var tunnelKeeper = new events.EventEmitter();
+
+tunnelKeeper.getTunnel = function (cb) {    // CAUTION: syncronous callback!
```

Comment:
Since this is internal code, I didn't see a reason to artificially delay `protoForConnection` and `Socket.prototype.connect` (the only callers of this helper) when a tunnel is already active. Reasonable enough?